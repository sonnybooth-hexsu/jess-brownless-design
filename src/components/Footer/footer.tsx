import React from "react"
import { Link } from "gatsby"
import { Linkedin, Instagram, Send } from "react-feather"
import logo from "../../assets/icons/logo.svg"
import containerClasses from "../../styles/container.module.css"

type FooterProps = {
  siteTitle: string
  productItems: {
    oculoplasticServices: {
      edges: Array<any>
    }
    aestheticServices: {
      edges: Array<any>
    }
    conditions: {
      edges: Array<any>
    }
  }
}

export const Footer = ({ siteTitle, productItems }: FooterProps) => (
  <>
    <div className="bg-Brand-green text-whiteBright">
      <div className={`${containerClasses.container} py-10`}>
        <div className="grid grid-cols-1 gap-6 mt-8 lg:grid-cols-12">
          <div className="grid grid-cols-12 col-span-12 gap-6">
            <div className="col-span-12 lg:col-span-6">
              <Link
                to="/"
                className="inline-block mb-4 text-lg uppercase md:text-3xl md:mb-0"
              >
                <img className="h-5" src={logo} alt={siteTitle} />
              </Link>
              <p className="my-4 uiRegular text-whiteWarm lg:pr-4">
                Elizabeth Hawkes is a Consultant Ophthalmic and Oculoplastic
                Surgeon, specialising in blepharoplasty surgery, eye health and
                advanced facial aesthetics.
              </p>
              <a
                target="_blank"
                rel="noopener noreferrer"
                href="https://www.instagram.com/dr_elizabethhawkes/"
              >
                <Instagram className="inline mr-2" />
              </a>
              <a
                target="_blank"
                rel="noopener noreferrer"
                className="ml-2"
                href="https://www.linkedin.com/in/elizabeth-hawkes-84410588/"
              >
                <Linkedin className="inline" />
              </a>
            </div>
          </div>
          <div className="col-span-12 mt-6 lg:mt-0 lg:col-span-3 xl:col-span-3">
            <nav>
              <h3 className="mb-2 text-sm font-bold tracking-wide">
                Oculoplastic Services
              </h3>
              <div className="mb-1 border-solid border-b-1"></div>
              <ol>
                {productItems?.oculoplasticServices?.edges.map(
                  ({ node }, id) => (
                    <li
                      className="pb-1 text-white transition-all duration-500 ease-in-out hover:underline hover:text-secondaryBrand-8"
                      key={`footerProductsLink-${id}`}
                    >
                      <Link
                        to={node?.childMarkdownRemark?.frontmatter?.path}
                        className="text-sm text-center"
                      >
                        {node?.childMarkdownRemark?.frontmatter?.title}
                      </Link>
                    </li>
                  )
                )}
              </ol>
            </nav>
          </div>
          <div className="col-span-12 lg:col-span-3 xl:col-span-3">
            <nav>
              <h3 className="mb-2 text-sm font-bold tracking-wide">
                Aesthetic Services
              </h3>
              <div className="mb-1 border-solid border-b-1"></div>
              <ol>
                {productItems?.aestheticServices?.edges.map(({ node }, id) => (
                  <li
                    className="pb-1 text-white transition-all duration-500 ease-in-out hover:underline hover:text-secondaryBrand-8"
                    key={`footerProductsLink-${id}`}
                  >
                    <Link
                      to={node?.childMarkdownRemark?.frontmatter?.path}
                      className="text-sm text-center"
                    >
                      {node?.childMarkdownRemark?.frontmatter?.title}
                    </Link>
                  </li>
                ))}
              </ol>
            </nav>
          </div>
          <div className="col-span-12 lg:col-span-3 xl:col-span-3">
            <nav>
              <h3 className="mb-2 text-sm font-bold tracking-wide">
                Eye Conditions
              </h3>
              <div className="mb-1 border-solid border-b-1"></div>
              <ol>
                {productItems?.conditions?.edges.map(({ node }, id) => (
                  <li
                    className="pb-1 text-white transition-all duration-500 ease-in-out hover:underline hover:text-secondaryBrand-8"
                    key={`footerProductsLink-${id}`}
                  >
                    <Link
                      to={node?.childMarkdownRemark?.frontmatter?.path}
                      className="text-sm text-center"
                    >
                      {node?.childMarkdownRemark?.frontmatter?.title}
                    </Link>
                  </li>
                ))}
              </ol>
            </nav>
          </div>
          <div className="col-span-12 mt-6 lg:mt-0 lg:col-span-3 xl:col-span-3">
            <h3 className="mb-2 text-sm font-bold">Stay up to date</h3>
            <div className="mb-8 border-solid border-b-1"></div>
            <form
              name="newsletter"
              method="post"
              data-netlify="true"
              data-netlify-honeypot="bot-field"
            >
              <div className="flex">
                <input type="hidden" name="form-name" value="newsletter" />
                <input type="hidden" name="bot-field" />
                <label className="flex flex-grow">
                  <span className="sr-only">Sign up</span>
                  <input
                    className="flex-grow pl-2 text-sm placeholder-white bg-transparent border-white border-solid lg:w-full border-b-1"
                    type="email"
                    placeholder="Email address"
                    id="newsletter-signup"
                    name="newsletter-signup"
                    required
                  />
                </label>
                <button className="p-3 bg-secondaryBrand-8">
                  <span className="sr-only">Apply for newsletter</span>
                  <Send />
                </button>
              </div>
            </form>
            <p className="mt-2 mb-6 text-xs text-gray-200">
              Please sign up to the newsletter if interested
            </p>
          </div>
        </div>
      </div>
    </div>
    <div className="py-2 text-white bg-mainBrand-10">
      <div
        className={`${containerClasses.container} px-6 mx-auto sm:px-8 md:px-12`}
      >
        <div className="flex justify-start">
          <p className="text-xs text-white">
            Made by{" "}
            <a
              target="_blank"
              rel="noopener noreferrer"
              className="text-xs text-white transition-all duration-500 ease-in-out hover:underline hover:text-secondaryBrand-8"
              href="https://www.instagram.com/jessicabrownless/?hl=en"
            >
              Jess Brownless Design
            </a>
          </p>
        </div>
      </div>
    </div>
  </>
)

export default Footer
