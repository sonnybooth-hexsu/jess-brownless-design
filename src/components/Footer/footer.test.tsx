/* eslint no-undef: 0 */

import React from "react"
import { render, cleanup } from "@testing-library/react"
import Footer from "./footer"

afterEach(() => {
  cleanup()
})

const productItems = {
  oculoplasticServices: {
    edges: [],
  },
  aestheticServices: {
    edges: [],
  },
  conditions: {
    edges: [],
  },
}

describe("<Footer />", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <Footer productItems={productItems} siteTitle="Test Site" />
    )

    expect(container).toMatchSnapshot()
  })
})
