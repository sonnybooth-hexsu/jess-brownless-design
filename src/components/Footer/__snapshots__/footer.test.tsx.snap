// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match snapshot 1`] = `
<div>
  <div
    class="bg-Brand-green text-whiteBright"
  >
    <div
      class="container py-10"
    >
      <div
        class="grid grid-cols-1 gap-6 mt-8 lg:grid-cols-12"
      >
        <div
          class="grid grid-cols-12 col-span-12 gap-6"
        >
          <div
            class="col-span-12 lg:col-span-6"
          >
            <a
              class="inline-block mb-4 text-lg uppercase md:text-3xl md:mb-0"
              href="/"
            >
              <img
                alt="Test Site"
                class="h-5"
                src="test-file-stub"
              />
            </a>
            <p
              class="my-4 uiRegular text-whiteWarm lg:pr-4"
            >
              Elizabeth Hawkes is a Consultant Ophthalmic and Oculoplastic Surgeon, specialising in blepharoplasty surgery, eye health and advanced facial aesthetics.
            </p>
            <a
              href="https://www.instagram.com/dr_elizabethhawkes/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="inline mr-2"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="20"
                  rx="5"
                  ry="5"
                  width="20"
                  x="2"
                  y="2"
                />
                <path
                  d="M16 11.37A4 4 0 1 1 12.63 8 4 4 0 0 1 16 11.37z"
                />
                <line
                  x1="17.5"
                  x2="17.51"
                  y1="6.5"
                  y2="6.5"
                />
              </svg>
            </a>
            <a
              class="ml-2"
              href="https://www.linkedin.com/in/elizabeth-hawkes-84410588/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="inline"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"
                />
                <rect
                  height="12"
                  width="4"
                  x="2"
                  y="9"
                />
                <circle
                  cx="4"
                  cy="4"
                  r="2"
                />
              </svg>
            </a>
          </div>
        </div>
        <div
          class="col-span-12 mt-6 lg:mt-0 lg:col-span-3 xl:col-span-3"
        >
          <nav>
            <h3
              class="mb-2 text-sm font-bold tracking-wide"
            >
              Oculoplastic Services
            </h3>
            <div
              class="mb-1 border-solid border-b-1"
            />
            <ol />
          </nav>
        </div>
        <div
          class="col-span-12 lg:col-span-3 xl:col-span-3"
        >
          <nav>
            <h3
              class="mb-2 text-sm font-bold tracking-wide"
            >
              Aesthetic Services
            </h3>
            <div
              class="mb-1 border-solid border-b-1"
            />
            <ol />
          </nav>
        </div>
        <div
          class="col-span-12 lg:col-span-3 xl:col-span-3"
        >
          <nav>
            <h3
              class="mb-2 text-sm font-bold tracking-wide"
            >
              Eye Conditions
            </h3>
            <div
              class="mb-1 border-solid border-b-1"
            />
            <ol />
          </nav>
        </div>
        <div
          class="col-span-12 mt-6 lg:mt-0 lg:col-span-3 xl:col-span-3"
        >
          <h3
            class="mb-2 text-sm font-bold"
          >
            Stay up to date
          </h3>
          <div
            class="mb-8 border-solid border-b-1"
          />
          <form
            data-netlify="true"
            data-netlify-honeypot="bot-field"
            method="post"
            name="newsletter"
          >
            <div
              class="flex"
            >
              <input
                name="form-name"
                type="hidden"
                value="newsletter"
              />
              <input
                name="bot-field"
                type="hidden"
              />
              <label
                class="flex flex-grow"
              >
                <span
                  class="sr-only"
                >
                  Sign up
                </span>
                <input
                  class="flex-grow pl-2 text-sm placeholder-white bg-transparent border-white border-solid lg:w-full border-b-1"
                  id="newsletter-signup"
                  name="newsletter-signup"
                  placeholder="Email address"
                  required=""
                  type="email"
                />
              </label>
              <button
                class="p-3 bg-secondaryBrand-8"
              >
                <span
                  class="sr-only"
                >
                  Apply for newsletter
                </span>
                <svg
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="22"
                    x2="11"
                    y1="2"
                    y2="13"
                  />
                  <polygon
                    points="22 2 15 22 11 13 2 9 22 2"
                  />
                </svg>
              </button>
            </div>
          </form>
          <p
            class="mt-2 mb-6 text-xs text-gray-200"
          >
            Please sign up to the newsletter if interested
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="py-2 text-white bg-mainBrand-10"
  >
    <div
      class="container px-6 mx-auto sm:px-8 md:px-12"
    >
      <div
        class="flex justify-start"
      >
        <p
          class="text-xs text-white"
        >
          Made by
           
          <a
            class="text-xs text-white transition-all duration-500 ease-in-out hover:underline hover:text-secondaryBrand-8"
            href="https://www.instagram.com/jessicabrownless/?hl=en"
            rel="noopener noreferrer"
            target="_blank"
          >
            Jess Brownless Design
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
